packages/core/src/Stores/client-store.js

```diff
@@ -1683,7 +1683,9 @@ export default class ClientStore extends BaseStore {
             runInAction(() => {
                 this.is_populating_account_list = false;
             });
-            const language = authorize_response.authorize.preferred_language;
+            const language =
+                new URLSearchParams(window.location.search).get('lang') ??
```

Comment:
what if language from url search param is not supported in deriv app?

Reply:
it will then be user's `preferred_language`

Reply:
getAllowedLanguages() in packages/translations/src/i18next/i18next.ts has supported languages.
here language can even be set if `?lang=SS` is being passed in the URL. right?

Reply:
@suisin-deriv we need to check whether the lang is supported by deriv app before redirecting to the url language.

Reply:
Ok let me check on this 🙏

Reply:
@shaheer-deriv if `lang=SS` is passed inside, it will take user's preferred language if the language is not supported.